Mathlib/Analysis/AbsoluteValue/Equivalence.lean

```diff
@@ -59,4 +59,109 @@ lemma eq_trivial_of_isEquiv_trivial [DecidablePred fun x : R ↦ x = 0] [NoZeroD
+      ← map_pow, ← map_inv₀, ← map_mul]
+    apply h
+    rw [map_mul, map_inv₀, map_pow, mul_inv_lt_iff₀ (pow_pos (by simp_all) _), one_mul]
+    exact lt_of_lt_of_le hx₀.2 <| one_le_pow₀ hv
+  obtain ⟨n, hn⟩ := exists_pow_lt_of_lt_one (w.pos hx₀.1) hw
+  linarith [this n]
+
+end LinearOrderedField
+
+variable {F : Type*} [Field F] {v w : AbsoluteValue F ℝ}
+
+open Real in
+/--
+If $v$ and $w$ are two real absolute values on a field $F$, $v$ is non-trivial, and $v(x) < 1$ if
+and only if $w(x) < 1$, then $\frac{\log (v(a))}{\log (w(a))}$ is constant for all $a ∈ F$
+with $1 < v(a)$.
+-/
+theorem log_div_eq_constant_of_abv_lt_one_iff (hv : v.IsNontrivial) (h : ∀ x, v x < 1 ↔ w x < 1) :
+    ∃ a, 1 < v a ∧ ∀ b, 1 < v b → (v b).log / (w b).log = (v a).log / (w a).log := by
+  let ⟨a, ha⟩ := hv.exists_abv_gt_one
+  refine ⟨a, ha, fun b hb₁ ↦ ?_⟩
+  by_contra! hb₂
+  wlog h_lt : (v b).log / (w b).log < (v a).log / (w a).log generalizing a b
+  · exact this b hb₁ a ha hb₂.symm <| lt_of_le_of_ne (not_lt.1 h_lt) hb₂.symm
+  have hwa := (v.one_lt_iff_of_lt_one_iff h _).1 ha
+  have hwb := (v.one_lt_iff_of_lt_one_iff h _).1 hb₁
+  rw [div_lt_div_iff₀ (log_pos hwb) (log_pos hwa), mul_comm (v a).log,
+    ← div_lt_div_iff₀ (log_pos ha) (log_pos hwa)] at h_lt
+  let ⟨q, ⟨hq₁, hq₂⟩⟩ := exists_rat_btwn h_lt
+  rw [← Rat.num_div_den q, Rat.cast_div, Rat.cast_intCast, Rat.cast_natCast] at hq₁ hq₂
+  rw [div_lt_div_iff₀ (log_pos ha) (by simp [q.den_pos]), mul_comm, ← log_pow, ← log_zpow,
+    log_lt_log_iff (pow_pos (by linarith) _) (zpow_pos (by linarith) _),
+    ← div_lt_one (zpow_pos (by linarith) _), ← map_pow, ← map_zpow₀, ← map_div₀] at hq₁
+  rw [div_lt_div_iff₀ (by simp [q.den_pos]) (log_pos hwa), mul_comm (w _).log,
+    ← log_pow, ← log_zpow, log_lt_log_iff (zpow_pos (by linarith) _) (pow_pos (by linarith) _),
+    ← one_lt_div (zpow_pos (by linarith) _), ← map_pow, ← map_zpow₀, ← map_div₀] at hq₂
+  exact not_lt_of_gt ((h _).1 hq₁) hq₂
+
+open Real in
+theorem exists_rpow_of_abv_one_lt_iff {v w : AbsoluteValue F ℝ} (hv : v.IsNontrivial)
+    (h : ∀ x, v x < 1 ↔ w x < 1) :
+    ∃ (t : ℝ) (_ : 0 < t), ∀ x, 1 < v x → w x ^ t = v x := by
+  obtain ⟨a, ha, hlog⟩ := log_div_eq_constant_of_abv_lt_one_iff hv h
+  refine ⟨(v a).log / (w a).log,
+    div_pos (log_pos ha) (log_pos ((v.one_lt_iff_of_lt_one_iff h a).1 ha)), fun b hb ↦ ?_⟩
+  simp_rw [← hlog b hb]
+  have := (v.one_lt_iff_of_lt_one_iff h b).1 hb
+  rw [div_eq_inv_mul, rpow_mul (w.nonneg _), rpow_inv_log (by linarith) (by linarith),
+    exp_one_rpow, exp_log (by linarith)]
+
+open Real in
+/--
+If $v$ and $w$ be two real absolute values on a field $F$, where $v$ is non-trivial, then $v$ and
+$w$ are equivalent if and only if $v(x) < 1$ exactly when $w(x) < 1$.
+-/
+theorem isEquiv_iff_abv_lt_one_iff {v : AbsoluteValue F ℝ} (w : AbsoluteValue F ℝ)
+    (hv : v.IsNontrivial) : w.IsEquiv v ↔ (∀ x, v x < 1 ↔ w x < 1) := by
```

Comment:
Ahah! So secretly all your lemmas are about equivalent valuations. What about you make all of them private, since `w.IsEquiv v` is the normal form of `∀ x, v x < 1 ↔ w x < 1`? Btw shouldn't it be
```suggestion
    (hv : v.IsNontrivial) : v.IsEquiv w ↔ ∀ x, v x < 1 ↔ w x < 1 := by
```
and do you really need `v.IsNontrivial`?